src/Autofac.Integration.ServiceFabric/ActorFactoryRegistration.cs

```diff
@@ -52,38 +53,40 @@ public ActorFactoryRegistration(Action<Exception> constructorExceptionCallback)
         {
             ActorRuntime.RegisterActorAsync<TActor>((context, actorTypeInfo) =>
             {
-                return new ActorService(
-                    context,
-                    actorTypeInfo,
-                    (actorService, actorId) =>
+                Func<ActorService, ActorId, ActorBase> actorFactory = (actorService, actorId) =>
+                {
+                    var lifetimeScope = container.BeginLifetimeScope(builder =>
                     {
-                        var lifetimeScope = container.BeginLifetimeScope(builder =>
-                        {
-                            builder.RegisterInstance(context)
-                                .As<StatefulServiceContext>()
-                                .As<ServiceContext>();
-                            builder.RegisterInstance(actorService)
-                                .As<ActorService>();
-                            builder.RegisterInstance(actorId)
-                                .As<ActorId>();
-                        });
-                        try
-                        {
-                            var actor = lifetimeScope.Resolve<TActor>();
-                            return actor;
-                        }
-                        catch (Exception ex)
-                        {
-                            // Proactively dispose lifetime scope as interceptor will not be called.
-                            lifetimeScope.Dispose();
+                        builder.RegisterInstance(context)
+                            .As<StatefulServiceContext>()
+                            .As<ServiceContext>();
+                        builder.RegisterInstance(actorService)
+                            .As<ActorService>();
+                        builder.RegisterInstance(actorId)
+                            .As<ActorId>();
+                    });
+                    try
+                    {
+                        var actor = lifetimeScope.Resolve<TActor>();
+                        return actor;
+                    }
+                    catch (Exception ex)
+                    {
+                        // Proactively dispose lifetime scope as interceptor will not be called.
+                        lifetimeScope.Dispose();
+
+                        ConstructorExceptionCallback(ex);
+                        throw;
+                    }
+                };
 
-                            ConstructorExceptionCallback(ex);
-                            throw;
-                        }
-                    },
-                    stateManagerFactory,
-                    stateProvider,
-                    settings);
+                return container.Resolve<ActorService>(
```

Comment:
It would be good to pass in the `Type` of the custom `ActorService` to the `RegisterActorAsync` method and use the `Resolve(Type, params Parameter[])` method to create the instance. That would allow different types of custom `ActorService` to be used for different actors. It would require casting the resolved service to `ActorService` so the `Type` passed in as a parameter should be tested that it inherits from `ActorService`.